microsoftgraph/microsoft-graph-explorer
# Task/js outdir

## Overview

* Species an output folder for the Typescript compiler
* Adds a file utility for copying html & css files into the output folder.

### Demo

N/A

### Notes

After specifying the output folder for the Typescript compiler it was noted that the compiler does not copy the accompany html & css files into the output folder(s). This utility program creates a folder structure similar to the one Typescript would have created and copies the html and css files into the relevant folder.

### Known Issues
Because of 
```
SystemJS.import('compiler-output/main-jit.js')
```
in  `index.html` an error will be thrown as the compiler would not have generated this file yet. As a consequence the developer has to reload the page.
**NB** This is only an issue the very first time the repo is set up on the developers' machine.

### Recommendations
I recommend that we move to `webpack` as a module loader because once we install our modules we do not have to tell it where the modules are as is the case with `SystemJS`. Explicitly telling `SystemJS` where to load modules from becomes excruciating really fast.

## Testing Instructions

* Pull these changes
* Run `npm start`
* Observe that there is a folder `compiler-output` with **html, css & js files**
